src/algorithms/pathfinders/path.ts

```diff
@@ -0,0 +1,33 @@
+import EdgeCost from '../models/edgecost';
+import Vertex from '../models/vertex';
+/**
+ * Path class, stores pathData and costs in a path
+ */
+export default class Path {
+    pathData: Array<{vertex: Vertex, edgecost: EdgeCost}>;
+    constructor() {
+        this.pathData = new Array<{vertex: Vertex, edgecost: EdgeCost}>();
+    }
+
+    add(vertex: Vertex, edgecost: EdgeCost): void {
+        this.pathData.push({vertex, edgecost});
+    }
+    addFront(vertex: Vertex, edgecost: EdgeCost): void {
+        this.pathData.unshift({vertex, edgecost});
+    }
+    evaluate(): number {
+        let pathCost = 0;
```

Comment:
This is a wonderful use case for reduction function :D with pathCost as an accumulator in this case.